pkg/kubernetes/clientbuilder.go

```diff
@@ -52,12 +65,30 @@ func (cb *ClientBuilder) Build(validate bool) (kubernetes.Interface, error) {
 	}
 
 	if validate {
-		timeout := int64(5)
-		_, err := clientSet.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{
-			TimeoutSeconds: &timeout,
+		ctx, cancel := context.WithCancel(ctx)
```

Comment:
Do we need another cancellable ctx here?
We're not explicitly cancelling it and the `defer` is invoked only after `wait.ExponentialBackoffWithContext` is done, which means we're not really cancelling.
It looks we rely entirely on the backoff timeout calculated internally inside `wait.ExponentialBackoffWithContext`
I think it's enough to pass the "parent" context to the `wait.ExponentialBackoffWithContext`.

Reply:
For me it's more a convention thing that I always open up a context if I propagate it downstream. As upstream we're using the request context, I found it better to create a new context and use this inside the exponentialbackoffwithcontext. 

I agree regarding necessity though: I will exclude it as per your request because its not strictly necessary.